### PR TITLE
Add WordCount template sample

### PR DESCRIPTION
**Pull request by davidcavazos**
Code sample for the [Creating templates](https://cloud.google.com/dataflow/docs/guides/templates/creating-templates) page.

### CODE REVIEW

File: dataflow/templates/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+Samples showing how to create and run an [Apache Beam][] on [Google Cloud Dataflow][].
+
+## Setup
+
+Setup instructions assume you have an active Google Cloud Project and with an associated billing account.
+The following instructions will help you prepare your development environment.
+
+1. Install [Cloud SDK][].
+1. Setup the Cloud SDK
```
Comment (lesv) [inline dataflow/templates/README.md]: If not already done

File: dataflow/templates/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+   ```bash
+   gcloud init
```
Comment (lesv) [inline dataflow/templates/README.md]: Well, that's nice, might there be more?  Create a project perhaps?  Enable some API's?

File: dataflow/templates/WordCount_metadata [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (lesv) [inline dataflow/templates/WordCount_metadata]: Should there be a license & copyright here?

File: dataflow/templates/pom.xml [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+  <version>1.0-SNAPSHOT</version>
+
+  <packaging>jar</packaging>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+    <beam.version>2.13.0</beam.version>
+
+    <bigquery.version>v2-rev20181104-1.27.0</bigquery.version>
```
Comment (lesv) [inline dataflow/templates/pom.xml]: That looks old

Comment (lesv) [reply dataflow/templates/pom.xml]: 1.77.0 appears recent

File: dataflow/templates/pom.xml [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+    <google-clients.version>1.27.0</google-clients.version>
+    <guava.version>20.0</guava.version>
```
Comment (lesv) [inline dataflow/templates/pom.xml]: really old

Comment (lesv) [reply dataflow/templates/pom.xml]: Check the other versions.

File: dataflow/templates/pom.xml [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+          <groupId>org.apache.beam</groupId>
+          <artifactId>beam-runners-flink_2.11</artifactId>
+          <version>${beam.version}</version>
+          <scope>runtime</scope>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <profile>
+      <id>spark-runner</id>
```
Comment (lesv) [inline dataflow/templates/pom.xml]: Should some of this stuff come from the beam BOM?

File: dataflow/templates/pom.xml [hunk 7]
```diff
@@ -0,0 +1,499 @@
...
+          <artifactId>guava-jdk5</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+
+    <dependency>
+      <groupId>joda-time</groupId>
+      <artifactId>joda-time</artifactId>
+      <version>${joda.version}</version>
+    </dependency>
```
Comment (lesv) [inline dataflow/templates/pom.xml]: If this isn't being using directly, many of our classes us a hybrid that uses joda for Java 7 and earlier, and java time for later.

File: dataflow/templates/pom.xml (hunk 5 above)
Comment (davidcavazos) [reply dataflow/templates/pom.xml]: Simplified pom file

**Pull request merged**